api/views/assets.py

```diff
@@ -123,19 +123,27 @@ def get_queryset(self):
 class AssetViewSet(ModelViewSet):
     serializer_class = AssetSerializer
     permission_classes = [IsAuthenticated]
+    queryset = models.Asset.objects
     authentication_classes = (FirebaseTokenAuthentication,)
     http_method_names = ["get"]
 
     def get_queryset(self):
         user = self.request.user
         query_filter = {}
-
         if not self.request.user.is_securityuser:
             asset_assignee = models.AssetAssignee.objects.filter(user=user).first()
-            query_filter = {"assigned_to": asset_assignee}
+            query_filter["assigned_to"] = asset_assignee
+            user_id = self.request.query_params.get("user_id")
+            if user_id:
+                if user.is_staff and models.User.objects.filter(id=user_id):
+                    query_filter["assigned_to"] = models.AssetAssignee.objects.filter(
+                        user=user_id
+                    ).first()
+                else:
+                    return self.queryset.none()
```

Comment:
When a non-staff user requests for assets assigned to a user, an empty payload is returned here. I would suggest a message informing them they are not authorised to complete the request . e.g 
````
else:
       raise APIException("Operation not permitted. You are not authorised to complete this request",
                           code=status.HTTP_401_UNAUTHORIZED)
````